src/vkex/Device.cpp

```diff
@@ -399,6 +417,40 @@ vkex::Result CDevice::InitializeQueueRequests()
   return vkex::Result::Success;
 }
 
+void CDevice::InitializeExtensionFeatures()
+{
+  m_create_info.app_p_next = m_create_info.p_next;
+  void* current_p_next = const_cast<void *>(m_create_info.p_next);
```

Comment:
The pnext handling here looks to me like the first extension added will have it's pnext pointing to the address of the create info pnext, which I think would be incorrect.

I think initializing current_p_next to nullptr would fix it, since you attach the chain to the create info at the end of this block.

Reply:
I was just thinking about this last night! I think I did a bad job naming `current_p_next`. This pointer is actually the _front_ of the `pNext` chain, not the back. So as I go through the features I might add, I just push them to the front. Whatever the app passed in gets pre-pended by these feature requests.

You can see that each per-feature block updates `current_p_next` to point to the current block, and the current block points to the previous one.

Anyway, I had forgotten that I wanted to rename this variable, and this is a great reminder.

How does `p_next_front` sound? or `p_next_top`?

Reply:
the idea being it's the "top" of the chain right? so if you expect create_info -> ext1 -> ext2 -> ext3 in the end, you'd be adding ext3, then 2, then 1, and then point create_info pnext at ext1. That's what I'm getting is the idea?

I think that would make sense, but I still think then the initializing is wrong here since you currently end up with create_info -> ext1 -> ext2 -> ext3 -> create_info.pNext; the last one is unnecessary and invalid.

As for the naming, I don't think it's a huge deal any which way personally; I agree that "front" or "top" is a little more clear than current though. I'd probably go with "front" myself, since that implies a list to me moreso than "top", and this is indeed a list I'd say.

Reply:
> create_info -> ext1 -> ext2 -> ext3 -> create_info.pNext; the last one is unnecessary and invalid.

Err, I think this is a case of me being too 'clever', which I hate because I think 'clever' code makes code unreadable. If the app doesn't have anything in the pNext pointer, then it should be null, and that's required to terminate the pNext chain. If the app _does_ have something in the pNext pointer, I can't throw it away, so I'm preserving it. Either way, I can retain the original pointer that was in create_info.p_next.

Reply:
Oh, I think I've been a bit of a moron and misunderstanding this a bit. This is in case pnext already  has something in it, I think?

Reply:
Ahhh, I see what you mean. Ya, it's already been populated (possibly just with a nullptr)